src/charm.py

```diff
@@ -1594,6 +1604,18 @@ def get_available_resources(self) -> Tuple[int, int]:
 
         return cpu_cores, allocable_memory
 
+    def on_deployed_without_trust(self) -> None:
+        """Blocks the application and returns a specific error message for deployments made without --trust."""
+        self.unit.status = BlockedStatus(
+            "Unauthorized access to k8s resources. Is the app trusted? See logs"
+        )
+        logger.error(
+            f"""
+            Access to k8s cluster resources is not authorized. This happens when RBAC is enabled and the deployed application was not trusted by the juju admin.
+            To fix this issue, run `juju trust {self._name} --scope=cluster` (or remove & re-deploy {self._name} with `--trust`)
+            """
+        )
+
```

Comment:
Please update PR description accordingly. It still mention `juju resolve` there (not-aplicable for blocked state).

Reply:
I think I had removed it, where do you see it mentioned?

Reply:
Oh good catch, thanks!